Springproofs (#11)
AAU-Dat/P10-Curdleproofs: report/src/sections/04-Approach.tex

In \autoref{fig:fold}(a) is a scheme function which simply pads the vector to the next power of two before running an~\gls{ipa}.
If one wanted to run current~\glspl{ipa}s on vector that are not a power of two, this would generally be the easiest way to achieve that.
Though, this defeats the attempt of lowering the proof size, as it would now correspond to running an~\gls{ipa} on the size of the next power of two.

\begin{figure*}[t]
    \centering
    \subfloat[\centering Padding Method]{{\includegraphics[width=0.43\textwidth]{figures/padding} }}%
    \qquad
    \subfloat[\centering Optimal Pre-Compression Method]{{\includegraphics[width=0.45\textwidth]{figures/fold} }}%
    \caption{Folding visualization as seen in the Springproofs paper}%
    \label{fig:fold}%
\end{figure*}

It is notable to mention that using the folding as shown in~\autoref{fig:fold}(b) results in the second recursive round being a size corresponding to a power of two.
This means that the rest of the protocol will run as a general~\gls{ipa}, without the actual need for splitting the vectors, which can also be seen in~\autoref{fig:schemefunc}.

> With this Springproofs in mind, we have made a modification to the~\gls{ipa} of Curdleproofs

Reviewer: ```suggestion
With the idea from Springproofs in mind, we have made a modification to the~\gls{ipa} of Curdleproofs
```

$d\gets d_1$

return $(B_C,B_D,\mathbf{\pi},c,d)$
    \end{lstlisting}
\end{framed}
\caption{Prover computation for CAAU-IPA in CAAUrdleproofs}
\label{fig:ipa-prover}
\end{figure}

First, we have step 1, which is the setup phase.

> It is done exactly the same way is in Curdleproofs.

Reviewer: ```suggestion
It is done exactly the same way as in Curdleproofs.
```

To ensure zero-knowledge, two blinding vectors for each commitment are constructed.
These are also given the properties, $(\mathbf{r}_C\times \mathbf{d}+\mathbf{r}_D\times \mathbf{c})=0$ and $\mathbf{r}_C\times\mathbf{r}_D=0$, ensuring the completeness of the protocol.

From the public input, hash values $\alpha,\beta$ are then computed.
These are used to ensure the soundness of the protocol.

The two vectors are then blinded and multiplied by the $\alpha$ hash to ensure the zero-knowledge and soundness, as well as $H=\beta H$.

Now, the recursive proof construction begins.

> As explained, at the start of the recursive round, the \texttt{while}-loop, we find the split the vectors, with $f(n)$ being the function from~\autoref{fig:schemefunc}.

Reviewer: ```suggestion
As explained, at the start of the recursive round, the \texttt{while}-loop, we find the split of the vectors, with $f(n)$ being the function from~\autoref{fig:schemefunc}.
```

\end{lstlisting}
\end{framed}
\caption{Verifier computation for CAAU-IPA in CAAUrdleproofs}
\label{fig:ipa-verifier}
\end{figure}

The changes to the verifier protocol are equivalent to the ones made to the prover protocol.
First, the vectors are divided into the two sets, $|T|,|S|$.
The verifier then retrieves the cross-product commitment update values, $L_{C,j},L_{D,j},R_{C,j},R_{D,j}$.
These are used for constructing a new commitment according to the fold made at round $i$.
The corresponding left and right side cross-product are multiplied by a challenge, $\gamma_j,\gamma_j^{-1}$, respectively.

> By this time, the C and D commitments are a commitment to the original commitments along with the folded commitment.

Reviewer: ```suggestion
By this time, the $C$ and $D$ commitments are a commitment to the original commitments along with the folded commitment.
```

$\mathbf{G,G'}$ are updated as in~\autoref{fig:ipa-prover} before the protocol updates $n$ to be the length of the newly constructed vectors.

As in the prover protocol, this is then repeated for $\log n$ round, after which the vectors have length $1$.

At the end of the protocol, the verifier now does its final check.
From the prover, it has retreived the folded down $c$ and $d$ vectors.
It therefore constructs commitments with these elements.

> So, it constructs $c\times G_1+cdH$, which is the structure of the C commitment as well as $d\times G'_1$, which is the structure of the D commitment.

Reviewer: ```suggestion
So, it constructs $c\times G_1+cdH$, which is the structure of the $C$ commitment as well as $d\times G'_1$, which is the structure of the $D$ commitment.
```
